undertow/WFLY-9417_deployment_start_behaviour.adoc

```diff
@@ -0,0 +1,74 @@
+= Title
+:author:            Stuart Douglas
+:email:             sdouglas@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WFLY-6493 changed the default behaviour of the server to queue requests while the server
+is starting, instead of just rejecting them (or potentially letting a request through
+to an incomplete deployments).
+
+This feature is about making this feature configurable, so that a user can choose
+between the current behaviour of queueing requests until the server has started,
+or between rejecting requests until the server has started.
+
+Note that neither of these selections will allow request processing to start
+until the server has completely started, which is an improvement on the original
+behaviour.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-6493
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-958
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Affected Projects or Components
+
+ * Undertow
+
+== Requirements
+
+=== Hard Requirements
+
+This feature will provide an attribute on the host called `queue-requests-on-start`, this attribute
+will default to `true`.
+
+If this attribute is true then there will be no deviation from the current behaviour, requests
+that arrive when the server is starting will be held until the server is ready.
+
+If this attribute is false then requests that arrive before the server has completely
+started will be rejected with the default response code.
+
+No matter the value of this attribute request processing will not start until the
+server is completely started, making it impossible for requests to be processed
+by an incomplete deployment.
+
+== Test Plan
+
+This is complex to test, as our existing test infrastructure is not set up to perform
+tests while the server is starting up. Such a test would also likely be fragile as
+it would be totally dependent on the amount of time the server takes to start.
+
+It is anticipated that automated testing cannot reasonably be done for this feature,
+and as such manual verification will be performed only.
```

Comment:
we have some basic tests for graceful shutdown in our internal testsuite, I think these could be extended to cover this using SCI

Reply:
This section might be updated as there have been created some simple tests in our internal TS. Not sure whether we should mention internal testsuites here or whether we should keep such information just in tests-plans...